text/orderer-v3.md

```diff
@@ -0,0 +1,394 @@
+the gRPC stream is an `Auth` message, and the rest are either `ConsensusRequest` or `SubmitRequest` messages.
+
+
+
+### The current authentication mechanism: Mutual TLS with TLS pinning 
+
+The Raft consensus library uses numerical identifiers to represent nodes taking part in the consensus protocol.
+When the Fabric runtime receives a request from the Raft library to send a message to a remote node, 
+it establishes a gRPC stream over mutual TLS with the remote node, and sends the message over the stream.
+Similarly, when a message is received from a remote node to a Raft orderer, the Fabric runtime locates the numerical identifier
+of the remote node that corresponds to the TLS certificate extracted from the TLS connection and then passes the message tagged with the numerical identifier.
+The conversion between the TLS certificate to the numerical identifier is done via a mapping that is found in the configuration blocks.
+
+Clearly, it is imperative that unauthorized parties will not be able to impersonate ordering service nodes.
+When performing in a Byzantine setting where malicious orderers might attempt impersonation, it is vital to authenticate which node
+sent the message. While it is possible to have each orderer sign every message it sends, it is inefficient.
+
+The current authentication mechanism of Fabric orderer-to-orderer communication relies on mutual TLS.
+More specifically, whenever an orderer wants to send a message to another orderer, it establishes a TLS connection to the remote orderer and the remote orderer identifies
+the connection initiator by its TLS certificate. This is a rather simple approach and its security entirely draws upon the native Go TLS runtime.
+
+```
+ __________       <orderer1.crt>      ___________
+|          |    ------------------>   |          |
+| orderer1 |    <------------------   | orderer2 |
+|          |      <orderer2.crt>      |          |
+|__________|                          |__________|
+```
+
+However, this approach has several drawbacks: 
+1. **TLS keys might not be stored in HSM**: The Go TLS stack doesn't provide HSM integration capabilities, and therefore TLS private keys of orderer nodes are more susceptible for theft. 
+A theft of a TLS private key means that an adversary may send messages on behalf of the victim node, which is catastrophic in a crash fault tolerant (CFT) setting.
+Although it is possible to override the `PrivateKey` reference of the `tls.Certifciate` with a signer implementation that communicates with an HSM, existing users which already have key management systems 
+   built will be reluctant to make the required changes.
+   
+
+
+2. **Mutual TLS authentication is challenging when combined with reverse proxies**: In case the organization hosting the orderer has a TLS terminating reverse proxy that routes ingress traffic into the organization,
+it is challenging for that organization to host an ordering service node that relies on mutual TLS authentication.
```

Comment:
Wouldn't the `ExportKeyingMaterial` approach mentioned below also has this limitation?

Reply:
I address that in the quote below:


> In case the responder node is located behind a TLS terminating reverse proxy, it can be configured to ignore the `session_binding` check,  but the check will be enabled by default.